pkg/reconciler/v1alpha1/pipelinerun/pipelinerun.go

```diff
@@ -441,20 +442,20 @@ func (c *Reconciler) updateTaskRunsStatusDirectly(pr *v1alpha1.PipelineRun) erro
 func (c *Reconciler) createTaskRun(logger *zap.SugaredLogger, rprt *resources.ResolvedPipelineRunTask, pr *v1alpha1.PipelineRun, storageBasePath string) (*v1alpha1.TaskRun, error) {
 	var taskRunTimeout = &metav1.Duration{Duration: 0 * time.Second}
 
-	if pr.Spec.Timeout != nil {
+	// If the value of the timeout is 0 for any resource, there is no timeout.
+	// It is impossible for pr.Spec.Timeout to be nil, since SetDefault always assigns it with a value.
+	if pr.Spec.Timeout.Duration != apisconfig.NoTimeoutDuration {
 		pTimeoutTime := pr.Status.StartTime.Add(pr.Spec.Timeout.Duration)
 		if time.Now().After(pTimeoutTime) {
 			// Just in case something goes awry and we're creating the TaskRun after it should have already timed out,
-			// set a timeout of 0.
+			// set a timeout of -1.
 			taskRunTimeout = &metav1.Duration{Duration: time.Until(pTimeoutTime)}
 			if taskRunTimeout.Duration < 0 {
-				taskRunTimeout = &metav1.Duration{Duration: 0 * time.Second}
+				taskRunTimeout = &metav1.Duration{Duration: -1 * time.Second}
```

Comment:
I'm a bit confused by this: it looks like the validation for PipelineRuns will not allow a Timeout that is < 0, I would expect the validation for TaskRuns to be the same? If so, that would mean that if we hit this case, the TaskRun will fail to be created 🤔 

Other ideas: could we set this to 1 second instead? Could we not create the TaskRun in this case?

Reply:
I will change it to a valid short time(1 sec) for the timed-out case of taskrun.